token-lending/program/src/processor.rs

```diff
@@ -102,23 +108,26 @@ fn process_init_lending_market(
     let rent = &Rent::from_account_info(next_account_info(account_info_iter)?)?;
     let token_program_id = next_account_info(account_info_iter)?;
 
+    assert_rent_exempt(rent, lending_market_info)?;
+    let mut lending_market = assert_uninitialized::<LendingMarket>(lending_market_info)?;
+    if lending_market_info.owner != program_id {
```

Comment:
Does this mean all `*.owner != program_id` checks can be removed?

Reply:
Removed all the ones I think are safe to remove in 2fb624f9f9cf05e28bae7a88151984d72a3d7fac and de2b5c5ecb750714f70eaf909eaf2d89bc5dce7c.

There are only a few left, and it might be possible to remove some. The lending market authority is used to transfer tokens from some reserves. If the token program will prevent this if the lending program doesn't own those reserves, then those checks aren't required, right?

Reply:
I would err towards keeping the owner checks actually because sometimes there are code paths in a program that actually don't mutate account state when you expect it to. Was mostly pointing out the case where we go from uninitialized to initialized where we definitely mutate account state. But again, the error message is nice. Without that, the error message is harder to debug because the runtime usually just complains that *some* account was written to which shouldn't have been

Reply:
Gotcha. I'll revert those then.